llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp

```diff
@@ -270,20 +357,70 @@ Register SPIRVGlobalRegistry::buildConstantFP(APFloat Val,
   return Res;
 }
 
+Register SPIRVGlobalRegistry::getOrCreateFloatCompositeOrNull(
+    Constant *Val, MachineInstr &I, SPIRVType *SpvType,
+    const SPIRVInstrInfo &TII, Constant *CA, unsigned BitWidth,
+    unsigned ElemCnt, bool ZeroAsNull) {
+  // Find a constant vector in DT or build a new one.
+  Register Res = DT.find(CA, CurMF);
+  bool IsNull = Val->isNullValue() && ZeroAsNull;
+  if (!Res.isValid()) {
+    // SpvScalConst should be created before SpvVecConst to avoid undefined ID
+    // error on validation.
+    // TODO: can moved below once sorting of types/consts/defs is implemented.
+    Register SpvScalConst;
```

Comment:
THe TODOs are also copied from `getOrCreateIntCompositeOrNull`.  I don't know when that work is to be scheduled or if there is already an issue tracking it. But  maybe there is more I can do  to keep from having to split into two seperate functions

Reply:
Ok, was curious because moving it a bit down and running tests didn't seem to fail, so was wondering if you had a repro.
Since it's from an old code, it might actually be out of date.